main/gatewayBT.cpp

```diff
@@ -955,11 +981,11 @@ void launchBTDiscovery(bool overrideDiscovery) {
         Log.trace(F("properties: %s" CR), properties.c_str());
         std::string brand = decoder.getTheengAttribute(p->sensorModel_id, "brand");
         std::string model = decoder.getTheengAttribute(p->sensorModel_id, "model");
-#    if ForceDeviceName
-        if (p->name[0] != '\0') {
-          model = p->name;
+        if (displayDeviceName || ForceDeviceName) {
+          if (p->name[0] != '\0') {
```

Comment:
If there is no name we should maybe go back to the model_id following @DigiH remark